### ISSUE TITLE
RPCAPI must give meaningful error messages

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#789]
PRs #460 and #494 implement validation of input data to RPCAPI. Some input data is fully controlled by the client (GUI), wheras other data is what the user has provided.

In one way it is good that the digest is validated, but the problem is that the error message from RPCAPI is not shown to the user in the GUI, and even if it did it would be hard to understand it for most users. I removed the last character in a valid hash and sent it to zmb:

```
"error":{
  "message":"Invalid method parameter(s).",
  "code":"-32602",
  "data":"/ds_info/0/digest: String does not match (?^ui:^[a-f0-9]{40}$|^[a-f0-9]{64}$)."
  }
```
It the digest is corrupted you do not get any help. I think that data that users can paste in GUI should be handled by RPCAPI in such a way that a meaningful error message can be returned. Of course, the GUI must also be prepared to display it for the user.

Another possibility of error in digest is that some digit is replaced. In that case the test is executed and there is some meaningful error messages for the user.

For the following set of data, the RPCAPI must return an error message that the GUI can present to the user and be helpful so that the user can correct it and have the domain being tested. The errors here are those that are returned when the cast will not even start.

* domain name
* name server name
* IP address
* DS key tag value
* DS digest

Edit: I updated the PR references above. (@mattias-p 2021-06-14)

**Comment by mattias-p** [comment]
Is it correct that this issue only concerns itself with the following methods?
* start_domain_test
* get_data_from_parent_zone
* get_ns_ips

Could you give examples of error messages you would like to see?

**Comment by matsduf** [comment]
> Is it correct that this issue only concerns itself with the following methods?
> 
>     * start_domain_test
>     * get_data_from_parent_zone
>     * get_ns_ips

Yes. Only those methods that directly depend on user data.

> Could you give examples of error messages you would like to see?

Domain name field:
* Illegal character in domain name
* Domain name too long
* Domain name label too long

Name server name sub field:
* Illegal character in domain name
* Domain name too long
* Domain name label too long

Name server IP address sub field:
* Not a valid IP address

DS key tag sub field:
* Wrong format

DS digest data sub field:
* Incorrect data size
* Illegal character in data
* Data size does not match digest type

This is based on the current GUI where e.g. hash type is a drop down list, which means that you cannot  select an illegal digest type. Error messages could be short if we add documentation for these things, e.g. in the FAQ.

It should be possible to translate the messages. One option would be if the RPCAPI sends error codes and GUI has a translation from the codes, also into English.

**Comment by blacksponge** [comment]
resolved by #853
